cmd/clusterctl/client/cluster/cert_manager.go

```diff
@@ -346,9 +345,14 @@ func (cm *certManagerClient) shouldUpgrade(desiredVersion string, objs, installO
 	// removes resources that are generated by the kubernetes API
 	// this is relevant if the versions are the same, because we compare
 	// the number of objects when version of objects are equal
-	objs = slices.DeleteFunc(objs, func(obj unstructured.Unstructured) bool {
-		return obj.GetKind() == "Endpoints" || obj.GetKind() == "EndpointSlice"
-	})
+	newObjs := []unstructured.Unstructured{}
+	for _, o := range objs {
+		if !(o.GetKind() == "Endpoints" || o.GetKind() == "EndpointSlice") {
+			newObjs = append(newObjs, o)
+		}
+	}
+	objs = newObjs
+
```

Comment:
Looking at this change, I realized that we don't want to modify any slice passed by the caller. Prior to this PR, we were merely ignoring some elements in the `objs` slice, but now we modify it.

I think we can address this by referencing `newObjs`, instead of `objs`, on lines 356 and 357.

(As an aside, I think the name `relevantObjs` might be more helpful than `newObjs`).

Reply:
thats a good point i changed the code because of side effects that i noticed in different methods when i modified the list.

Reply:
in practice, i think it's fine that endpoints and endpoint slices aren't removed because they will be garbage collected.